### ISSUE TITLE
Oxleas HL integration

### ISSUE THREAD

**Issue by iteles** [dwyl/hq#355]
Requirements:
- Additional user group
- Copy & styling differences
- Resources might be just content changes
- MW is meeting with them on 13th June
    - TBC whether SLaM will host on Azure

+ [x] Make a list of what would need to be done to decouple HL from ePJS
+ [x] Book meeting with Mollie to review scope
+ [x] Provide estimate for decoupling
+ [x] Provide estimate for additional user groups and styling differences

**Proposed start date: 17th July 2017, 1 sprint, 1 developer**

**Comment by katbow** [comment]
For parents being able to add content on behalf of their children:

* How is the parent/child relationship confirmed? Would this be through the clinician to say that X is the parent of Y? Or admin? My main concern would be someone who is not the parent being able to connect and write in data for that person. 

Once the connection exists, there could be an extra input field for any data entry that says `Posting for: Myself/my child`.  If the parent is posting for their child, we can take the id from the carer relationship table & insert for the `caring_id`. 

What I am uncertain about is how this would work for editing. An individual goal is restricted at the moment to the current user, so if a parent attempted to view/edit a single goal they wouldn't have access. It may be possible to allow editing if the current user id is from either the child or the parent, but then the problem is still with displaying all the goals of the parent vs the child. 

If editing needs to be included, then we can:
* have a different view for parents vs children in viewing all goals
* parents edit from the overview (similar to the clinician view of SU in HL), but this would still require  different views/logic between the parents and clinicians. 

**In summary:**
It should be relatively simple for parents to add new goals/coping strategies/tracking data for their child, ~4h to 1d. Adding editing functionality as well for their child's content may be more complex, as it would involve changing much of the logic for how we get & display both the full list & individual goals/strategies, and creating new views to differentiate between either parent/child or parent/clinician.

**Comment by iteles** [comment]
From K:
Here's a summary of the reduced priorities for Oxleas:
1 - Create Basic Admin Dashboard so Oxleas can add everyone and assign roles
2 - Reskim - Change logo/ banner, potentially change site colours although there's some wiggle room on that, a lot of content changes to all the guidance notes, terms of service, privacy statement, Get Support Page, all references to South London & Maudsley etc.
3 - New - 'About Me'.  Screen of free text boxes which both the SU and/or Clinician can complete with information about the SU.  Needs to be editable for both user groups
4 - New - Medications - V similar to About Me - screen of text boxes which only the Clinician can complete and edit - need to be viewable to SU and parent
5 - Allow Clinician to make diary entries on the SU's diary (not edit existing ones).

We can talk about it in more detail tomorrow, but that all feels pretty manageable.  Depending on the level of complexity it adds, it would be good to include teachers as an additional user group - they would need a caseload view as the clinicians have and would be able to view goals, coping strategies, about me and meds.  If it is a big piece of work then we can probably drop this, but as About Me and meds are being built new, and permissions being set up, it might not be too much additional work to set Teachers up at the same time and this would then add quite a bit of value.

**Comment by iteles** [comment]
In progress with @RobStallion
